extensions/resource-deployment/src/interfaces.ts

```diff
@@ -35,24 +35,85 @@ export interface ResourceTypeOptionValue {
 	displayName: string;
 }
 
-export interface DeploymentProvider {
-	title: string;
-	dialog: DialogInfo;
+export interface DialogDeploymentProvider extends DeploymentProvider {
+	dialog: DialogInfoTypes;
+}
+
+export interface WizardDeploymentProvider extends DeploymentProvider {
+	wizard: WizardInfo;
+}
+
+export interface NotebookDeploymentProvider extends DeploymentProvider {
 	notebook: string | NotebookInfo;
-	downloadUrl: string;
+}
+
+export interface WebPageDeploymentProvider extends DeploymentProvider {
 	webPageUrl: string;
-	wizard: WizardInfo;
+}
+
+export interface DownloadDeploymentProvider extends DeploymentProvider {
+	downloadUrl: string;
+}
+
+export interface CommandDeploymentProvider extends DeploymentProvider {
+	command: string;
+}
+
+export function instanceOfDialogDeploymentProvider(obj: any): obj is DialogDeploymentProvider {
+	return obj && 'dialog' in obj;
+}
+
+export function instanceOfWizardDeploymentProvider(obj: any): obj is WizardDeploymentProvider {
+	return obj && 'wizard' in obj;
+}
+
+export function instanceOfNotebookDeploymentProvider(obj: any): obj is NotebookDeploymentProvider {
+	return obj && 'notebook' in obj;
+}
+
+export function instanceOfWebPageDeploymentProvider(obj: any): obj is WebPageDeploymentProvider {
+	return obj && 'webPageUrl' in obj;
+}
+
+export function instanceOfDownloadDeploymentProvider(obj: any): obj is DownloadDeploymentProvider {
+	return obj && 'downloadUrl' in obj;
+}
+
+export function instanceOfCommandDeploymentProvider(obj: any): obj is CommandDeploymentProvider {
+	return obj && 'command' in obj;
+}
+
+export interface DeploymentProvider {
 	requiredTools: ToolRequirementInfo[];
 	when: string;
 }
 
+export type DeploymentProviderTypes = DialogDeploymentProvider | WizardDeploymentProvider | NotebookDeploymentProvider | WebPageDeploymentProvider | DownloadDeploymentProvider | CommandDeploymentProvider;
```

Comment:
>DeploymentProviderTypes [](start = 12, length = 23)

nit: but normally you wouldn't pluralize these. Just DeploymentProviderType is fine. And TBH I'd even just say DeploymentProvider is fine (we already know this is a type so that's kind of redundent)